connect/manifest.ts

```diff
@@ -0,0 +1,46 @@
+import axios from 'axios';
+
+export interface WebManifest {
+  icons?: { src: string; sizes?: string }[];
+}
+
+// Fetch the web manifest
+async function getManifestFile(url: string) {
+  try {
+    const response = await axios.get<WebManifest>(`${url}/manifest.json`);
+    return response.data;
+  } catch (err) {
+    const response = await axios.get<WebManifest>(`${url}/manifest.webmanifest`);
+    if (response.data) {
+      return response.data;
+    }
+  }
+}
+
+export async function getAppIconFromWebManifest(url: string): Promise<string> {
```

Comment:
this is a good candidate for unit test coverage.

it's highly likely the function will be extended in the future to parse different manifest files, so unit tests will be appropriate here